shopinvader_product_template_multi_link/models/shopinvader_product.py

```diff
@@ -42,16 +45,21 @@ def _get_shopinvader_variant_link(self, link_type):
                     # Set only the "main" shopinvader variant
                     for shopinvader_variant in binding.shopinvader_variant_ids:
                         if shopinvader_variant.main:
-                            res.append(shopinvader_variant.id)
+                            res |= shopinvader_variant
                             break
         return res
 
-    @api.multi
     def _compute_shopinvader_link(self):
         for record in self:
-            record.cross_selling_ids = record._get_shopinvader_variant_link(
-                "cross_sell"
+            record.cross_selling_ids = record._get_linked_shopinvader_variants(
+                self.env.ref(
+                    "product_template_multi_link."
```

Comment:
if these types are deleted this is going to be broken.
Options: 
1. use raise_if_not_exists=False and compute nothing if not found
2. add `cross_sell_type_id` and `up_sell_type_id` to the backend to make them configurable

In both cases, if they are no found or set nothing is computed (and maybe we log something).

Reply:
I'm in favour of 1. `use raise_if_not_exists=False and compute nothing if not found`